src/backend/distributed/utils/resource_lock.c

```diff
@@ -272,6 +272,28 @@ LockShardListMetadata(List *shardIntervalList, LOCKMODE lockMode)
 	}
 }
 
+/* LockPlacementListMetadata takes locks on the metadata of all shards in
+ * shardPlacementList to prevet concurrent placement changes.
```

Comment:
super minor: comment should start on second line, two empty lines between functions